src/crepe/system/InputSystem.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "facade/SDLContext.h"
+#include "types.h"
+#include "util/OptionalRef.h"
+
+#include "System.h"
+
+namespace crepe {
+
+class Camera;
+class Button;
+class Transform;
+
+/**
+ * \brief Handles the processing of input events like mouse and keyboard interactions.
+ *
+ * This system processes events such as mouse clicks, mouse movement, and keyboard
+ * actions. It is responsible for detecting interactions with UI buttons and
+ * passing the corresponding events to the registered listeners.
+ */
+class InputSystem : public System {
+public:
+	using System::System;
+
+	/**
+     * \brief Updates the system, processing all input events.
+     * This method processes all events and triggers corresponding actions.
+     */
+	void update() override;
+
+private:
+	//! Reference to the currently active camera.
+	OptionalRef<Camera> curr_cam_ref;
```

Comment:
Je slaat hier `OptionalRef<Camera> curr_cam_ref` op, maar het is helemaal niet nodig om die op te slaan. Je gebruikt deze reference alleen in de update methode, dus ik snap niet waarom je de references op zou slaan (qua mij betreft moet je Components ook zo min mogelijk opslaan).